state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1328,12 +1331,14 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		tx_id: <T::TargetChain as Chain>::DepositDetails,
 	) -> DispatchResult {
 		let account_id = T::AccountRoleRegistry::ensure_broker(origin)?;
-		TaintedTransactions::<T, I>::insert(
-			account_id,
-			tx_id,
+		TxExpiresAt::<T, I>::mutate(
 			<frame_system::Pallet<T>>::block_number()
```

Comment:
I'd suggest making it such that block numbers are grouped into 'hourly' buckets. The exact time is not important. 

Also we can use append:

```rust
TxExpiresAt::<T, I>::append(
	// Expiry scheduled between 1 and 2 hours in the future:
	((<frame_system::Pallet<T>>::block_number() % 600) + 2) * 600,
	(account_id.clone(), tx_id.clone())
);
```

Reply:
Why should we do this? 🤔 Is it not also bad for brokers (for e.x) to not reliable know when something expires?

Reply:
Actually yeah I can't remember why I thought this would be a good idea - never mind. (Although using append is still a better than mutate.)

Reply:
Agree 👍